File: java-backend/src/main/java/com/imagesearch/model/entity/FolderShare.java
package com.imagesearch.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

/**
 * FolderShare entity for folder sharing permissions.
 *
 * Represents sharing relationships between users and folders.
 * Supports different permission levels (view, edit, etc.).
 */
@Entity
@Table(name = "folder_shares", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"folder_id", "shared_with_user_id"})
})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FolderShare {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "folder_id", nullable = false)
    private Folder folder;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "shared_with_user_id", nullable = false)
    private User sharedWithUser;

    @Column(nullable = false)
    private String permission = "view";

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: dotnet-backend/ImageSearch.Api/Models/Entities/FolderShare.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ImageSearch.Api.Models.Entities;

[Table("folder_shares")]
public class FolderShare
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("id")]
    public long Id { get; set; }

    [Required]
    [Column("folder_id")]
    public long FolderId { get; set; }

    [Required]
    [Column("owner_id")]
    public long OwnerId { get; set; }

    [Required]
    [Column("shared_with_user_id")]
    public long SharedWithUserId { get; set; }

    [Required]
    [MaxLength(50)]
    [Column("permission")]
    public string Permission { get; set; } = "view";

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    [ForeignKey("FolderId")]
    public virtual Folder Folder { get; set; } = null!;

    [ForeignKey("OwnerId")]
    public virtual User Owner { get; set; } = null!;

    [ForeignKey("SharedWithUserId")]
    public virtual User SharedWithUser { get; set; } = null!;
}
